src/services/submissionService.ts

```diff
@@ -118,7 +118,23 @@ function convertToDbSubmission(submission: gqlType.SubmissionInput):
 export const addSubmission = async (submissionInput: gqlType.SubmissionInput): 
     Promise<dbSchema.Submission> => {
     try {
-        const dbSubmission = convertToDbSubmission(submissionInput)
+        if (!submissionInput.googleMapsUrl.trim() || !submissionInput.healthcareProfessionalName.trim()) {
+            throw new Error('Submission fields cannot be empty strings.')
+        }
+
+        const nonNullLanguages = submissionInput.spokenLanguages
+            .filter(lang => lang !== null) as gqlType.SpokenLanguageInput[]
```

Comment:
Null is actually different then undefined. We should check for both. 
You can likely use `!!lang` for the check